### PR TITLE
fix(metrics): expose completed download retention

### LINKED ISSUE #2122: Expose unambiguous download counter semantics
## Goal

Expose separate session and persisted completed-download totals without breaking v1 consumers, and establish `in_session` and `persisted` as the naming convention for API v2's unambiguous completed-count contract.

## Background

The legacy `tracker_core_persistent_torrents_downloads_total` counter is process-scoped when persistence is disabled but historical across restarts when persistent completed statistics are enabled. The public metric and REST `completed` documentation imply an always-persisted lifetime total, while the v1 response gives clients no retention-mode signal.

## Scope

- Retain the legacy REST field and metric identifier/value semantics for compatibility; correct their descriptions and document deprecation.
- Add v1 `completed_in_session`, `completed_persisted`, and `completed_persisted_enabled` fields.
- Maintain distinct tracker-core in-session and persisted views; export the persisted Prometheus metric only while persistent completed statistics are enabled.
- Derive REST persistence availability from validated configuration, not from numeric values or service presence.
- Review and extend the existing REST stats contract tests; add direct metrics endpoint coverage and top-level integration coverage when necessary to prove configuration-to-endpoint behavior.
- Add regression tests for persistence-free reset, persisted restoration, enabled zero values, export omission, and v1 deserialization compatibility.
- Record an ADR defining retention names, lifecycle, deprecation, and the additive v1-to-v2 migration.
- Manually verify both configurations and record exact local commands, requests, responses, and outcomes in the issue-local evidence artifact.

## Out of Scope

- Removing, renaming, or changing legacy conditional values before API v2.
- API v2 implementation, listener-topology changes, database schema or migration changes, and persistence-configuration redesign.

## Acceptance Criteria

- Legacy values remain compatible but accurately documented and deprecated.
- In-session values reset per process and persisted values are seeded from, then updated after successful writes to, the database aggregate.
- REST clients distinguish disabled persistence only through `completed_persisted_enabled`; Prometheus omits the persisted metric while disabled.
- REST server contract tests cover the expanded stats endpoint and metrics endpoint; add a top-level integration test when needed to prove the configuration-to-endpoint contract.
- The REST composition root supplies validated persistence capability; older JSON payloads remain deserializable.
- Tests cover both lifecycle modes, zero, export presence/absence, and existing listener topology.
- A repository-wide ADR and mandatory local verification evidence are completed before closure.

## References

- Prerequisite: #2107
- Earlier REST API v2 deferral refined by this additive v1 bridge: #999
- Full specification: `docs/issues/open/2122-expose-unambiguous-download-counter-semantics/ISSUE.md`

### PR DESCRIPTION
**Pull request by josecelano**
## Summary

Expose explicit `in_session` and `persisted` completed-download counters while retaining the legacy v1 REST field and Prometheus metric as deprecated compatibility views.

- Add additive v1 `completed_in_session`, `completed_persisted`, and `completed_persisted_enabled` fields with backward-compatible deserialization.
- Omit the persisted Prometheus metric when persistent completed statistics are disabled; preserve a zero-valued sample when enabled.
- Add a repository-wide ADR, tracker-core/REST contract regressions, and local SQLite manual-verification evidence.

## Verification

- Mandatory pre-commit gate passed.
- Full pre-push suite passed: nightly format/check/docs and all stable workspace tests.
- M1-M3 completed locally using persistence-free and SQLite configurations.

Closes #2122

### CODE REVIEW

File: docs/issues/open/2122-expose-unambiguous-download-counter-semantics/manual-verification.md [hunk 1]
```diff
@@ -12,73 +12,111 @@ HTTP requests.
 
 ## Local Environment
 
-- Operating system: {value}
-- Tracker command: {exact command}
-- Working directory: {path}
-- Configuration source: {environment variable or file path}
-- Configuration: {redacted TOML or link to an issue-local redacted fixture}
+- Operating system: Linux
+- Tracker command: `cargo run -- --config {CONFIG_PATH}`
```
[the remark this answers is not in the document]
Comment (josecelano) [reply docs/issues/open/2122-expose-unambiguous-download-counter-semantics/manual-verification.md]: Corrected the Local Environment tracker command to use `TORRUST_TRACKER_CONFIG_TOML_PATH`, matching the recorded M1 and M2 commands.

**Comment by josecelano** [comment]
ACK 1899d730a4a3a9e7ae6c90c781b15b59fd74a0b9
